### PR TITLE
Added dotnet_version 8 to both linux and windows function apps

### LINKED ISSUE #23637: Function App - Missing dotnet 8.0 for both Windows and Linux
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Terraform Version

1.6.2

### AzureRM Provider Version

3.77.0

### Affected Resource(s)/Data Source(s)

azurerm_linux_function_app, azurerm_windows_function_app

### Terraform Configuration Files

```hcl
provider "azurerm" {
  features {}
}

resource "azurerm_linux_function_app" "linux-example" {
  name                = "xxxx"
  resource_group_name = "xxxx"
  location            = "xxxx"

  storage_account_name       = "xxxx"
  storage_account_access_key = "xxxx"
  service_plan_id            = azurerm_service_plan.linux-example

  site_config {
    application_stack {
      dotnet_version              = "8.0" # currently only allows 3.1, 6.0, 7.0
    }
  }
}

resource "azurerm_windows_function_app" "windows-example" {
  name                = "dotnet8-example-windows-function-app"
  resource_group_name = "example-resources"
  location            = "Australia East"

  storage_account_name       = "xxxx"
  storage_account_access_key = "xxxx"
  service_plan_id            = azurerm_service_plan.windows-example

  site_config {
    application_stack {
      dotnet_version              = "8.0" # currently only allows v3.0, v4.0, v6.0 and v7.0
      use_dotnet_isolated_runtime = true
    }
  }
}

resource "azurerm_service_plan" "windows-example" {
  name                = "example-windows-app-service-plan"
  resource_group_name = "xxxx"
  location            = "xxxx"
  os_type             = "Windows"
  sku_name            = "Y1"
}

resource "azurerm_service_plan" "linux-example" {
  name                = "example-linux-app-service-plan"
  resource_group_name = "xxxx"
  location            = "xxxx"
  os_type             = "Linux"
  sku_name            = "Y1"
}
```

### Debug Output/Panic Output

```shell
N/A
```

### Expected Behaviour

Deploy with (Windows|Linux)FxVersion = "dotnet|8.0"

### Actual Behaviour

Error, as `8.0` (Linux) or `v8.0` (Windows) is not an allowed value currently.

### Steps to Reproduce

```sh
terraform plan

Error: expected site_config.0.application_stack.0.dotnet_version to be one of ["3.1" "6.0" "7.0"], got 8.0
Error: expected site_config.0.application_stack.0.dotnet_version to be one of ["v3.0" "v4.0" "v6.0" "v7.0"], got 8.0
```

### Important Factoids

_No response_

### References

https://learn.microsoft.com/en-us/azure/azure-functions/functions-versions?tabs=isolated-process%2Cv4&pivots=programming-language-csharp#tabpanel_2_v4_isolated-process

### PR DESCRIPTION
**Pull request by AdamCoulterOz**
Fixes #23637 - adds support for dotnet 8 to both linux and windows function apps

### CODE REVIEW

**Review by stephybun — CHANGES_REQUESTED** [review]
Thanks for this PR @AdamCoulterOz. This looks fine but just to be sure and since this version is in preview, could you please add an acceptance test where we set `dotnet_version` to `8.0`?

File: internal/services/appservice/helpers/function_app_schema.go [hunk 1]
```diff
@@ -989,7 +989,7 @@ type ApplicationStackLinuxFunctionApp struct {
 }
 
 type ApplicationStackWindowsFunctionApp struct {
-	DotNetVersion         string `tfschema:"dotnet_version"`              // Supported values `3.1`, `6` and `7`
+	DotNetVersion         string `tfschema:"dotnet_version"`              // Supported values `v3.0`, `v4.0`, `v6.0`, `v7.0` and `v8.0`
```
Comment (ilmax) [inline internal/services/appservice/helpers/function_app_schema.go:992]: Should this be 3.1 like everywhere else?

**Comment by AdamCoulterOz** [comment]
> Thanks for this PR @AdamCoulterOz. This looks fine but just to be sure and since this version is in preview, could you please add an acceptance test where we set `dotnet_version` to `8.0`?

I will, I haven't had a chance yet. I'll do it in the next few days.

**Comment by AdamCoulterOz** [comment]
@stephybun or @jackofallops can I please ask you to take a quick look at this? Thanks! 😀

**Review by jackofallops — APPROVED** [review]
Thanks @AdamCoulterOz - This LGTM now 👍

**Pull request merged**